sled-agent/src/services.rs

```diff
@@ -954,6 +954,31 @@ impl ServiceManager {
         self.inner.sled_mode
     }
 
+    /// Returns the sled's identifier
+    fn sled_id(&self) -> Uuid {
+        self.inner
+            .sled_info
+            .get()
+            .expect("sled agent not started")
+            .config
+            .sled_id
+    }
+
+    /// Returns the metrics queue for the sled agent if it is running.
+    fn maybe_metrics_queue(&self) -> Option<&MetricsRequestQueue> {
+        self.inner.sled_info.get().map(|info| &info.metrics_queue)
+    }
+
+    /// Returns the metrics queue for the sled agent.
+    fn metrics_queue(&self) -> &MetricsRequestQueue {
+        &self
+            .inner
+            .sled_info
+            .get()
+            .expect("Sled agent should have started")
+            .metrics_queue
```

Comment:
Maybe delegate this to: `self.maybe_metrics_queue().expect("Sled agent should have started")`?